internal/xds/translator/cluster.go

```diff
@@ -77,6 +78,30 @@ func buildXdsCluster(args *xdsClusterArgs) *clusterv3.Cluster {
 		cluster.TransportSocket = args.tSocket
 	}
 
+	for i, ds := range args.settings {
+		if ds.TLS != nil {
+			// for upstreamTLS , a fixed sni can be used. use auto_sni otherwise
+			// https://www.envoyproxy.io/docs/envoy/latest/faq/configuration/sni#faq-how-to-setup-sni:~:text=For%20clusters%2C%20a,for%20trust%20anchor.
+			socket, err := buildXdsUpstreamTLSSocketWthCert(ds.TLS)
+			if err != nil {
+				// TODO: Log something here
+				return nil
+			}
+			if args.proxyProtocol != nil {
```

Comment:
this should probably live outside the loop ?

Reply:
It can't live outside the loop because the proxy protocol socket wraps the socket created inside the loop.

Reply:
Done.